test_harness/benchmarks/categories/src/hebench_benchmark_category.cpp

```diff
@@ -103,78 +103,95 @@ bool PartialBenchmarkCategory::validateResult(IDataLoader::Ptr dataset,
+                    throw std::invalid_argument(IL_LOG_MSG_CLASS("Buffer in outputs is not large enough to contain the expected output: 'outputs'."));
+                }
+
+                std::uint64_t count = (*it_truths)->size / IDataLoader::sizeOf(data_type);
+                void *p_truth       = (*it_truths)->p;
+                void *p_output      = (*it_outputs)->p; // single output
+
+                // validate the results
+                switch (data_type)
+                {
+                case hebench::APIBridge::DataType::Int32:
+                    is_valid = hebench::Utilities::Math::almostEqual(reinterpret_cast<const std::int32_t *>(p_truth),
+                                                                     reinterpret_cast<const std::int32_t *>(p_output),
+                                                                     count,
+                                                                     0.01);
+                    break;
+
+                case hebench::APIBridge::DataType::Int64:
+                    is_valid = hebench::Utilities::Math::almostEqual(reinterpret_cast<const std::int64_t *>(p_truth),
+                                                                     reinterpret_cast<const std::int64_t *>(p_output),
+                                                                     count,
+                                                                     0.01);
+                    break;
+
+                case hebench::APIBridge::DataType::Float32:
+                    is_valid = hebench::Utilities::Math::almostEqual(reinterpret_cast<const float *>(p_truth),
+                                                                     reinterpret_cast<const float *>(p_output),
+                                                                     count,
+                                                                     0.01);
+                    break;
+
+                case hebench::APIBridge::DataType::Float64:
+                    is_valid = hebench::Utilities::Math::almostEqual(reinterpret_cast<const double *>(p_truth),
+                                                                     reinterpret_cast<const double *>(p_output),
+                                                                     count,
+                                                                     0.01);
+                    break;
+
+                default:
+                    retval = false;
+                    break;
+                } // end switch
+
+                // In case retval is set to false, it will break the for loop
+                retval = retval && is_valid.empty();
             } // end for
-            if (is_valid.size() > MaxErrorPrint)
-                ss << ", ...";
-            throw std::runtime_error(ss.str());
+
+            if (!retval)
+            {
+                std::stringstream ss;
+                ss << "Elements not within 1% of each other, " << is_valid.size() << std::endl
```

Comment:
As a request, can we change this line to:

```cpp
ss << "Result component " << component_index << "; elements not within 1% of each other, " << is_valid.size() << std::endl
```

This will give feedback on which result component failed the validation test. Use indices ( `component_index` ) instead of iterators as mentioned earlier.